enhancements/monitoring/multi-tenant-alerting.md

```diff
@@ -517,6 +524,17 @@ of the monitoring components.
 
 N/A
 
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+The validating webhook is configured with `failurePolicy: Ignore` to not block
```

Comment:
@wking - this will hopefully address your concern that we don't have a situation like https://github.com/openshift/cluster-monitoring-operator/pull/1203

Reply:
that means invalid data is accepted when the operator is down? That's not good.

Reply:
Note: you can restrict the webhook to create and update.

Reply:
The upstream webhook implementation is already restricted to create and update.

Reply:
What's the issue with CVO exactly? Why should it be able to create or update these objects without the webhook?

Reply:
To be honest, when I wrote this section, I copied-pasted what CMO does currently for PrometheusRules resources (which are the only "monitoring" resources that go through a webhook right now) and I didn't put more thoughts into it. But I agree that you have good points @sttts.

The issue with CVO was [BZ1949711](https://bugzilla.redhat.com/show_bug.cgi?id=1949711). Whenever someone deleted the openshift-monitoring namespace, CVO couldn't recreate the monitoring resources and got stuck. The "workaround" was to change the `failurePolicy` setting. Now I don't remember the exact root cause and I realize that deleting manually the validating webhook might have unblock CVO as well. The other reason why we chose `failurePolicy: Ignore` was to avoid blocking other operators when the prometheus operator is down (during upgrades for instance since it's a single replica).

In retrospect the reasons aren't solid and we should rather deploy a highly-available service for monitoring resources validation webhook.